consensus/src/experimental/pipeline_phase.rs

```diff
@@ -29,10 +49,20 @@ impl<T: StatelessPipeline> PipelinePhase<T> {
         Self { rx, tx, processor }
     }
 
+    pub async fn exhaust_requests_non_blocking(&mut self) {
+        while self.rx.next().now_or_never().is_some() {
+            hint::spin_loop()
+        }
+    }
+
     pub async fn start(mut self) {
         // main loop
         while let Some(req) = self.rx.next().await {
-            let resp = self.processor.process(req).await;
+            let ResponseWithInstruction { resp, instruction } = self.processor.process(req).await;
+            match instruction {
+                Instruction::Ok => {}
+                Instruction::Clear => self.exhaust_requests_non_blocking().await,
```

Comment:
I don't see this is used anywhere? when do we need this?

Reply:
We need this if there is an execution failure. We want the execution phase to empty the req channel because later reqs will fail anyway.

Reply:
we could just inspect the resp to see if it's BlockNotFound? don't see we need this for every phases

Reply:
The current abstraction prevents the phases (through `process` function) from interacting with the channels directly. So I add this extra function (that for now only execution phase will be using, but I think persisting phase might also need this?).